### PR TITLE
Fix alias HEAD requests

### PR DESCRIPTION
**Pull request by jasontedor**
Alias HEAD requests incorrectly return a content-length header of 0. This commit addresses this by removing the special handling for alias HEAD requests, and just relying on the general mechanism that exists for handling HEAD requests in the REST layer.

Relates #21125

### CODE REVIEW

**Review by javanna — APPROVED** [review]
left a question, LGTM otherwise

File: core/src/main/java/org/elasticsearch/rest/action/admin/indices/RestAliasesExistAction.java [hunk 1]
```diff
@@ -1,87 +0,0 @@
...
-import static org.elasticsearch.rest.RestRequest.Method.HEAD;
-import static org.elasticsearch.rest.RestStatus.NOT_FOUND;
-import static org.elasticsearch.rest.RestStatus.OK;
-
-public class RestAliasesExistAction extends BaseRestHandler {
-    public RestAliasesExistAction(Settings settings, RestController controller) {
-        super(settings);
-        controller.registerHandler(HEAD, "/_alias/{name}", this);
-        controller.registerHandler(HEAD, "/{index}/_alias/{name}", this);
-        controller.registerHandler(HEAD, "/{index}/_alias", this);
```
Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/rest/action/admin/indices/RestAliasesExistAction.java]: I don't think this endpoint does anything, wouldn't it always 404 or am I missing something?

Comment (javanna) [inline core/src/main/java/org/elasticsearch/rest/action/admin/indices/RestAliasesExistAction.java]: you are right. I have no idea why we had that. makes no sense.

**Comment by jasontedor** [comment]
Thanks @javanna.

**Pull request merged**